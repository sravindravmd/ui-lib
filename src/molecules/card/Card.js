import React from 'react'
import PropTypes from 'prop-types'
import ExpandableCard from './expandable'
import CarouselCard from './carousel'
import FullAccessCard from './fullaccess'
import RootCategoryCard from './rootcategory'
import QuickAccessCard from './quickaccess'
import HierarchyTreeCard from './hierarchytree'
import CardItems from './carditem'

const CardTypes = {
  expandable: ExpandableCard,
  carousel: CarouselCard,
  general: FullAccessCard,
  withActionItem: FullAccessCard,
  rootCategory: RootCategoryCard,
  quickAccess: QuickAccessCard,
  hierarchytree: HierarchyTreeCard,
}

export const Card = ({ variant, heading, children, ...props }) => {
  const Component = CardTypes[variant] && CardTypes[variant]

  if (!Component) {
    return null
  }
  return (
    <>
      <Component {...props} variant={variant} heading={heading}>
        {children}
      </Component>
    </>
  )
}

export const CardItem = CardItems

Card.propTypes = {
  variant: PropTypes.oneOf([
    'expandable',
    'carousel',
    'general',
    'withActionItem',
    'rootCategory',
    'quickAccess',
    'hierarchytree',
  ]),
  heading: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
}
