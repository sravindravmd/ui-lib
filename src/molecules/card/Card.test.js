import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Card } from './Card'

afterEach(cleanup)

describe('<Card/>', () => {
  it('renders Card with variant:expandable correctly ', () => {
    const { container } = render(<Card variant='expandable' heading='Card' />)
    expect(container).toMatchSnapshot()
  })
})
