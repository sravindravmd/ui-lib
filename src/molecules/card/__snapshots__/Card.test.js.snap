// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders Card with variant:expandable correctly  1`] = `
<div>
  <div
    class="sc-bdfBwQ fWZGSR"
  >
    <div
      class="sc-dlfnbm jYeWQn"
    >
      <div
        class="sc-hKgILt eLIWMT"
      >
        Card
      </div>
      <div
        class="sc-gsTCUz dAxUvf"
        data-testid="downarrow"
      >
        <svg
          fill="none"
          height="7"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0.519346 0.527549C0.807911 0.236494 1.27804 0.235307 1.56808 0.5249L6.47801 5.42743C6.7666 5.71557 6.76778 6.18285 6.48065 6.47245C6.19209 6.76351 5.72196 6.76469 5.43192 6.4751L0.521985 1.57257C0.233403 1.28443 0.232222 0.81715 0.519346 0.527549Z"
            fill="#121213"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M11.4807 0.527549C11.1921 0.236494 10.722 0.235307 10.4319 0.5249L5.52199 5.42743C5.2334 5.71557 5.23222 6.18285 5.51935 6.47245C5.80791 6.76351 6.27804 6.76469 6.56808 6.4751L11.478 1.57257C11.7666 1.28443 11.7678 0.81715 11.4807 0.527549Z"
            fill="#121213"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCssSg boTWkf"
    />
  </div>
</div>
`;
